include/olg_scheduler/scheduler.hpp

```diff
@@ -135,17 +142,12 @@ class Event : private cavl::Node<Event<TimePoint>>
     /// (otherwise it will keep firing in a tight loop).
     /// If the event needs to be re-scheduled automatically, it must be done before the user callback is invoked,
     /// because the user may choose to cancel the event from the callback.
-    virtual void execute(const Arg<TimePoint>& args) = 0;
+    virtual void execute(const Arg<TimePoint>& args, Tree& tree) = 0;
 
 private:
-    cavl::Tree<Event>&       tree_;
```

Comment:
B/c of latest CAVL, we don't need anymore `tree_` for the `cancel()`. So, now even less footprint of an event.